src/test/java/com/Incubatormsrv/Incubatormsrv/IncubatorMsrvApplicationTests.java

```diff
@@ -1,13 +1,204 @@
+
+		ResponseEntity<CurrentWeatherResponse> response = currentWeatherController.currentWeatherGet(cityName, numOfDays);
+
+		assertEquals(HttpStatus.OK, response.getStatusCode());
+		assertEquals(currentWeatherResponse, response.getBody());
+	}
 
 	@Test
-	void contextLoads() {
+	public void testCurrentWeatherGet_Exception() throws Exception {
+		String cityName = "Pretoria";
+		BigDecimal numOfDays = BigDecimal.valueOf(4);
+		WeatherApiException weatherApiException = new WeatherApiException("An error occurred while getting weather data");
+
+		Mockito.when(weatherService.getWeather(cityName, numOfDays)).thenThrow(weatherApiException);
+		Mockito.when(threadExecutor.getExecutor()).thenReturn(Executors.newSingleThreadExecutor());
+
+		Exception exception = Assert.assertThrows(ExecutionException.class, () -> currentWeatherController.currentWeatherGet(cityName, numOfDays));
+
+		Assert.assertTrue(exception.getCause() instanceof WeatherApiException);
+		Assert.assertEquals(weatherApiException.getMessage(), exception.getCause().getMessage());
 	}
 
+	@Test
+	public void testGetWeather_Success() throws Exception {
+
+		WeatherResponse weatherResponseMocked = new WeatherResponse();
+
+		File jsonFile = new File("src/test/java/mocks/weather.json");
+		String Path = jsonFile.getAbsolutePath();
+		String jsonFileData = FileUtils.readFileToString(jsonFile,"UTF-8");
+
+		weatherResponseMocked = new Gson().fromJson(jsonFileData,WeatherResponse.class);
+
+		Response<WeatherResponse> response = Response.success(weatherResponseMocked);
+
+		wireMock.stubFor(get(urlPathMatching("/weather"))
+				.withQueryParam("q", equalTo("Pretoria"))
+				.withQueryParam("format", equalTo("json"))
+				.withQueryParam("num_of_days", equalTo(String.valueOf(BigDecimal.valueOf(4))))
+				.willReturn(aResponse()));
+
+		doReturn(response).when(weatherApi).weatherGet("Pretoria", "json",  BigDecimal.valueOf(4));
+		WeatherResponse result = weatherService.getWeather("Pretoria", BigDecimal.valueOf(4));
+
+		Assert.assertEquals(weatherResponseMocked, result);
+	}
+
+	@Test
+	public void testGetWeather_Failure() throws Exception {
+		String errorMessage = "An error occurred while getting weather data";
+		Response<WeatherResponse> response = Response.error(400, ResponseBody.create(MediaType.parse("text/plain"), errorMessage));
+
+		wireMock.stubFor(get(urlPathMatching("/weather"))
+				.withQueryParam("q", equalTo("Pretoria"))
+				.withQueryParam("format", equalTo("json"))
+				.withQueryParam("num_of_days", equalTo(String.valueOf(BigDecimal.valueOf(4))))
+				.willReturn(aResponse().withStatus(400).withBody(errorMessage)));
+
+		doReturn(response).when(weatherApi).weatherGet("Pretoria", "json",  BigDecimal.valueOf(4));
+		Exception exception = Assert.assertThrows(IOException.class, () -> weatherService.getWeather("Pretoria", BigDecimal.valueOf(4)));
+		Assert.assertEquals(errorMessage, exception.getMessage());
+	}
+	@After
```

Comment:
Group this with the top of the class unit test configuration related code for readability so that I am logically able to see the "lifecycle" of the unit tests.